pkg/stub/handler.go

```diff
@@ -1,102 +1,119 @@
 package stub
 
 import (
-	"context"
+	"time"
 
 	"github.com/aerogear/mobile-developer-console/pkg/apis/aerogear/v1alpha1"
 	"github.com/aerogear/mobile-developer-console/pkg/mobile"
-	"github.com/operator-framework/operator-sdk/pkg/sdk"
-	log "github.com/sirupsen/logrus"
+	"github.com/labstack/gommon/log"
 	"k8s.io/api/core/v1"
 )
 
-func NewHandler(mobileAppRepo mobile.MobileClientRepo) sdk.Handler {
-	return &Handler{
-		mobileClientRepo: mobileAppRepo,
+func WatchSecrets(namespace string, secretsCRUDL mobile.SecretsCRUDL, mobileAppRepo mobile.MobileClientRepo) {
+	getWatchInterface := secretsCRUDL.Watch(namespace)
+	watchInterface, err := getWatchInterface()
+	if err != nil {
+		return
 	}
-}
-
-type Handler struct {
-	// Fill me
-	mobileClientRepo mobile.MobileClientRepo
-}
-
-func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	switch o := event.Object.(type) {
-	case *v1.Secret:
-		secret := o
-
-		if !isValidSecret(secret) {
-			return nil
+	events := watchInterface.ResultChan()
+	defer func() {
+		if watchInterface != nil {
+			watchInterface.Stop()
 		}
-
-		labels := secret.GetLabels()
-		clientId := labels["clientId"]
-
-		app, err := h.mobileClientRepo.ReadByName(clientId)
-		if err != nil {
-			log.Errorf("failed to read mobile client by name %v", clientId)
-			return err
+	}()
+	willHandle := false
+	for {
+		select {
+		case _, ok := <-events:
+			if !ok {
+				watchInterface.Stop()
+				watchInterface, err = getWatchInterface()
+				if err != nil {
+					return
+				}
+				events = watchInterface.ResultChan()
+				continue
+			}
+			if willHandle {
+				continue
+			}
+			willHandle = true
+			timer := time.NewTimer(time.Second)
+			go func(timer *time.Timer) {
+				<-timer.C
+				willHandle = false
```

Comment:
this probably should be set to `true` after the next line is finished execution? otherwise there is a chance the next line could be invoked in parallel? Also I am not sure if this need to executed in the background? why not just execute it inside the loop? This way we don't need to use the `willHandle` flag.

Reply:
It is kind of a hack which makes sure, that in case there are e.g. 10 events at the same time (noticed that this happens e.g. when server starts) we will execute the handler function only once.

Reply:
maybe try use the `ResourceVersion` field in the `ListOptions` and see if it can help?